### ISSUE TITLE
Question/Docs/Readme: how do packs fit into things?

### ISSUE THREAD

**Issue by renschler** [dapperlabs/nba-smart-contracts#141]
I'm not sure if this is the right place to post this (please close this issue if it's out of place) but after walking through the Readme, Topshot contract, and Market contract, it's not clear to me how/where packs are implemented. Since this is a core aspect of NBA topshots I thought it might make sense to shed more light on them in the readme, but I definitely could be wrong, I'm just getting into things.

Are moments minted before a pack actually sells? Or does purchasing a pack trigger a random selection of moments to be minted?

Does the process of buying a pack fall under the Topshot Market contract (ie would Market.MomentPurchase events include pack purchases?)

I did see one of the lib/go files, there was a reference to PackNFT, but I couldn't find any references to PackNFT anywhere else.

**Comment by HouseOfHufflepuff** [comment]
Hello Renschler. Thanks for posting this issue. Packs are off-chain in NBA Top Shot. I will update the readme with this info, but let me answer your questions specifically. NFL-All-Day did an onchain pack implementation if you are looking for a reference.

`Are moments minted before a pack actually sells? Or does purchasing a pack trigger a random selection of moments to be minted?`
Moments are minted prior to the pack sale. They are not minted on-demand. The packing process randomizes moments into packs. The purchase path randomizes which pack a collector receives when they purchase.

`Does the process of buying a pack fall under the Topshot Market contract (ie would Market.MomentPurchase events include pack purchases?)`
Packs are entirely off-chain on NBA Top Shot. They are a collection of moments which are on-chain.

`I did see one of the lib/go files, there was a reference to PackNFT, but I couldn't find any references to PackNFT anywhere else.`
I will take a look at this, am not familiar off the top of my head. We have experimented with on-chain packs, but this experimentation has yet to make its way into product.
